source/neuropod/bindings/java/src/main/java/com/uber/neuropod/NeuropodTensorAllocator.java

```diff
@@ -39,68 +48,91 @@ protected NeuropodTensorAllocator(long handle) {
      * @param tensorType the tensor type
      * @return the created NeuropodTensor
      */
-    NeuropodTensor create(ByteBuffer byteBuffer, long[] dims, TensorType tensorType) {
-        return null;
+    public NeuropodTensor create(ByteBuffer byteBuffer, long[] dims, TensorType tensorType) {
+        if (!SUPPORTED_TENSOR.contains(tensorType)) {
+            throw new NeuropodJNIException("Unsupported tensor type: " + tensorType.name());
+        }
+        if (!byteBuffer.isDirect() || byteBuffer.order() != ByteOrder.nativeOrder() || byteBuffer.isReadOnly()) {
+            switch (tensorType) {
+                case INT32_TENSOR:
+                    return create(byteBuffer.asIntBuffer(), dims);
+                case INT64_TENSOR:
+                    return create(byteBuffer.asLongBuffer(), dims);
+                case FLOAT_TENSOR:
+                    return create(byteBuffer.asFloatBuffer(), dims);
+                case DOUBLE_TENSOR:
+                    return create(byteBuffer.asDoubleBuffer(), dims);
+            }
+        }
+        return createTensorFromBuffer(byteBuffer, dims, tensorType);
     }
 
     /**
      * Create a NeuropodTensor based on given LongBuffer, dims array. The created tensor
      * will have tensor type INT64_TENSOR.
      * <p>
-     * Will not trigger a copy if the buffer is a direct buffer in native order and the backend of
-     * the allocator does not have alignment requirement. Otherwise it will trigger a copy.
+     * Will trigger a copy.
```

Comment:
Why are we always triggering a copy?

Can you create an issue for us to optimize this (and other methods)?

Reply:
There is no way for a FloatBuffer/DoubleBuffer/IntBuffer... converting back to ByteBuffers without a copy, so here we have to trigger a copy.

Reply:
If the user does not want to trigger a copy then they have to pass direct byte buffer directly.
Do you think it is a good idea to name these methods differently to remind the different behavior of these create methods( Will or will not trigger a copy)
For example:
NeuropodTensor tensorFromMemory(ByteBuffer buffer, long [] dims, TensorType type);
NeuropodTensor copy_from(FloatBuffer buffer, long [] dims);